model-engine/model_engine_server/db/base.py

```diff
@@ -50,17 +58,17 @@ def get_engine_url(env: Optional[str] = None, read_only: bool = True, sync: bool
         if infra_config().cloud_provider == "azure":
             client = SecretClient(
                 vault_url=f"https://{os.environ.get('KEYVAULT_NAME')}.vault.azure.net",
-                credential=ManagedIdentityCredential(
-                    client_id=os.getenv("AZURE_KEYVAULT_IDENTITY_CLIENT_ID")
-                ),  # uses a different managed identity than the default
+                credential=DefaultAzureCredential(),
             )
             db = client.get_secret(key_file).value
             user = os.environ.get("AZURE_IDENTITY_NAME")
-            password = (
-                DefaultAzureCredential()
-                .get_token("https://ossrdbms-aad.database.windows.net/.default")
-                .token
+            token = DefaultAzureCredential().get_token(
```

Comment:
qq do you know at what frequency azure credentials refresh/if there's an overlap where both old and new credentials are valid, and if there's any sort of behavior where LLM Engine might keep some "old session" open that uses old credentials? 

e.g. suppose at time T the return value for get_token() changes from old_token to new_token, at which point does old_token expire, and also when's the next time get_token() would change?

mainly asking to see what work is needed for my use case